Makefile

```diff
@@ -145,65 +190,65 @@ build/flake8.mlos_core.${CONDA_ENV_NAME}.build-stamp: $(MLOS_CORE_PYTHON_FILES)
 
 .PHONY: pytest
 pytest: conda-env build/pytest.${CONDA_ENV_NAME}.build-stamp
 
-build/pytest.mlos_core.${CONDA_ENV_NAME}.needs-build-stamp: build/conda-env.${CONDA_ENV_NAME}.build-stamp
-build/pytest.mlos_core.${CONDA_ENV_NAME}.needs-build-stamp: $(MLOS_CORE_PYTHON_FILES) conftest.py setup.cfg
-build/pytest.mlos_core.${CONDA_ENV_NAME}.needs-build-stamp:
-	# Update the PYTEST_MODULES list to include mlos_core.
-	$(eval PYTEST_MODULES += mlos_core)
-	echo "PYTEST_MODULES: $(PYTEST_MODULES)"
-	touch $@
+pytest-mlos-core: build/pytest.mlos_core.${CONDA_ENV_NAME}.needs-build-stamp
+pytest-mlos-bench: build/pytest.mlos_bench.${CONDA_ENV_NAME}.needs-build-stamp
+pytest-mlos-viz: build/pytest.mlos_viz.${CONDA_ENV_NAME}.needs-build-stamp
 
-# Run the mlos_bench target update after mlos_core target update.
-build/pytest.mlos_bench.${CONDA_ENV_NAME}.needs-build-stamp: build/pytest.mlos_core.${CONDA_ENV_NAME}.needs-build-stamp
-build/pytest.mlos_bench.${CONDA_ENV_NAME}.needs-build-stamp: build/conda-env.${CONDA_ENV_NAME}.build-stamp
-build/pytest.mlos_bench.${CONDA_ENV_NAME}.needs-build-stamp: $(MLOS_BENCH_PYTHON_FILES) conftest.py setup.cfg
-build/pytest.mlos_bench.${CONDA_ENV_NAME}.needs-build-stamp:
-	# Update the PYTEST_MODULES list to include mlos_bench.
-	$(eval PYTEST_MODULES += mlos_bench)
-	echo "PYTEST_MODULES: $(PYTEST_MODULES)"
-	touch $@
+build/pytest.mlos_core.${CONDA_ENV_NAME}.needs-build-stamp: $(MLOS_CORE_PYTHON_FILES) $(MLOS_CORE_CONF_FILES)
+build/pytest.mlos_core.${CONDA_ENV_NAME}.needs-build-stamp: PYTEST_MODULE := mlos_core
+
+build/pytest.mlos_bench.${CONDA_ENV_NAME}.needs-build-stamp: $(MLOS_BENCH_PYTHON_FILES) $(MLOS_BENCH_CONF_FILES)
+build/pytest.mlos_bench.${CONDA_ENV_NAME}.needs-build-stamp: PYTEST_MODULE := mlos_bench
 
-# Run the mlos_viz target update after mlos_bench target update.
-build/pytest.mlos_viz.${CONDA_ENV_NAME}.needs-build-stamp: build/pytest.mlos_bench.${CONDA_ENV_NAME}.needs-build-stamp
-build/pytest.mlos_viz.${CONDA_ENV_NAME}.needs-build-stamp: build/conda-env.${CONDA_ENV_NAME}.build-stamp
-build/pytest.mlos_viz.${CONDA_ENV_NAME}.needs-build-stamp: $(MLOS_VIZ_PYTHON_FILES) conftest.py setup.cfg
-build/pytest.mlos_viz.${CONDA_ENV_NAME}.needs-build-stamp:
-	# Update the PYTEST_MODULES list to include mlos_viz.
-	$(eval PYTEST_MODULES += mlos_viz)
-	echo "PYTEST_MODULES: $(PYTEST_MODULES)"
+build/pytest.mlos_viz.${CONDA_ENV_NAME}.needs-build-stamp: $(MLOS_VIZ_PYTHON_FILES) $(MLOS_VIZ_CONF_FILES)
+build/pytest.mlos_viz.${CONDA_ENV_NAME}.needs-build-stamp: PYTEST_MODULE := mlos_viz
+
+# Invividual package test rules (for tight loop dev work).
+# Skip code coverage tests for these.
+PYTEST_COMMON_PREREQS := build/conda-env.${CONDA_ENV_NAME}.build-stamp
+PYTEST_COMMON_PREREQS += $(FORMAT_PREREQS)
+PYTEST_COMMON_PREREQS += $(PYTEST_CONF_FILES)
+
+build/pytest.%.${CONDA_ENV_NAME}.needs-build-stamp: $(PYTEST_COMMON_PREREQS)
+	conda run -n ${CONDA_ENV_NAME} pytest $(PYTEST_EXTRA_OPTIONS) $(PYTEST_MODULE)
```

Comment:
Individual `make pytest-mlos-core` `pytest-mlos-bench` etc. targets that can be used to run things in a tighter loop locally if desired.